### PR TITLE
DOC-326: Add a section on detaching to channels docs

### PR DESCRIPTION
**Pull request by m-hulbert**
> _NOTE TO REVIEWERS_ - please **do not review PRs in the `DRAFT` state**, as the PR may change substantially before it is ready to review. Thanks.

## Description

This PR adds a section to the realtime channel documentation on detaching from channels. See [JIRA](https://ably.atlassian.net/browse/DOC-326) for further information. 

## Review

View the [channels](https://ably-docs-pr-1262.herokuapp.com/realtime/channels) documentation to review the PR.

### CODE REVIEW

File: content/realtime/channels.textile [hunk 1]
```diff
@@ -813,6 +814,55 @@ Previously registered listeners can be removed individually or all together.
     stateChangeListener.cancel();
 ```
 
+
+h3(#channel-detach). Detaching from a channel
+
+A client can detach from a channel so that they no longer receive any messages published to that channel. Detaching is different to unsubscribing from a channel as "@unsubscribe()@":#unsubscribe is a client-side operation, meaning Ably is unaware of whether or not a client has unsubscribed from the channel. Messages will continue to be streamed to the client until "@detach()@":#detach is called.
```
Comment (marklewin) [inline content/realtime/channels.textile:820]: Suggest referring to a client as "it" to make it clear that we're talking about a bit of code rather than a company that pays for our services?

```suggestion
A client can detach from a channel so that it no longer receives any messages published to that channel. Detaching is different to unsubscribing from a channel as "@unsubscribe()@":#unsubscribe is a client-side operation, meaning Ably is unaware of whether or not a client has unsubscribed from the channel. Messages will continue to be streamed to the client until "@detach()@":#detach is called.
```

Comment (marklewin) [reply content/realtime/channels.textile:820]: Perhaps we can simplify this sentence, for example:

"A client can detach from a channel so that it no longer receives any messages published to that channel. Detaching is different to unsubscribing from a channel because "@unsubscribe()@":#unsubscribe is a client-side operation. The Ably platform does not know that a client has unsubscribed and will continue to stream messages to that client until "@detach()@":#detach is called."

File: content/realtime/channels.textile [hunk 2]
```diff
@@ -813,6 +814,55 @@ Previously registered listeners can be removed individually or all together.
...
+
+bc[flutter]. channel.detach();
+final stateChangeListener = channel
+  .on(ably.ChannelEvent.detached)
+  .listen((ably.ChannelStateChange state) {
+    print('detached from the channel ${channel.name}');
+  }
+);
+
+A channel will automatically close when there are no more realtime clients attached to it and approximately one minute has passed since the last client detached and since the last message was published to the channel.
```
Comment (marklewin) [inline content/realtime/channels.textile:864]: Is this correct? I'm thinking that, for a message to be published, a client must be attached to the channel to do the publishing. Therefore, the last client leaving will always be later than the last message being published? 🤔

**Review by marklewin — CHANGES_REQUESTED** [review]

File: content/realtime/channels.textile (hunk 1 above)
Comment (m-hulbert) [reply content/realtime/channels.textile]: Good point on the "it" and "client" point. I've taken your suggestion for the rewrite as it's far less wordy!

File: content/realtime/channels.textile (hunk 2 above)
Comment (m-hulbert) [reply content/realtime/channels.textile:864]: This is true as far as I'm aware, yes. You can always publish by REST so this can occur after you no longer have any realtime attachments. 

Found the [FAQ](https://faqs.ably.com/how-does-ably-count-peak-channels) I read this in for reference.

Comment (paddybyers) [reply content/realtime/channels.textile:864]: Correct, REST publishes can take place without attachment. (And also equivalently, realtime publishes via a transient attach - technically they are attaches under the hood, but clients don't see that - they just call `publish()`).

Comment (m-hulbert) [reply content/realtime/channels.textile:864]: Thanks for the confirmation, Paddy.

**Review by marklewin — APPROVED** [review]
LGTM!
